spec/controllers/review_controller_spec.rb

```diff
@@ -137,6 +137,74 @@
       end
     end
 
+    describe "POST request change after 2i review" do
+      let(:user) { create(:user) }
+      let(:reviewer_user) { create(:user) }
+
+      let(:step_by_step_page) do
+        create(
+          :draft_step_by_step_page,
+          review_requester_id: user.uid,
+          status: "submitted_for_2i"
+        )
+      end
+
+      before do
+        step_by_step_page.update_attributes(:status => 'in_review', :reviewer_id => reviewer_user.uid)
+      end
+
+      it "can be accessed by users with GDS editor, Unreleased feature and 2i reviewer permissions" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature", "2i reviewer"]
+        post :request_change_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response).to redirect_to step_by_step_page_path(step_by_step_page)
+      end
+
+      it "cannot be accessed by users with only GDS editor permissions" do
```

Comment:
And rather than testing where users only have one permission, I think we could have more confidence in our tests if we're testing for the absence of a required permission. For example:

```ruby
it "cannot be accessed by users missing the GDS editor permission" do
  stub_user.permissions = required_permissions - "GDS Editor"
  # ...etc
```

I'd even be tempted to take it a step further with something like this, as all the fail cases are identical:

```ruby
it "cannot be accessed by users missing any of the required permissions" do
  required_permissions.each.do |required_permission|
    stub_user.permissions = required_permissions - required_permission
    # ...etc
  end
end
```

Reply:
I really like this idea.  Implementing it as an additional commit for the other routes too.